### PR TITLE
Consolidated Requests

### PR DESCRIPTION
**Pull request by LaughDonor**
Consolidated all the requests to a centralized object. #22 
Cleaned up some logic and added comments.

### CODE REVIEW

**Comment by LaughDonor** [comment]
I've deprecated the `createDocumentWithId` function since `createDocument` can create an ID if a document name isn't given in the path. Goes with the consolidation. Let me know your thoughts @grahamearley.

**Comment by grahamearley** [comment]
@LaughDonor Awesome. Haven't had a change to do any testing with this one yet, but after a quick read-through it looks like another great PR. I'll get back to you with some more notes after I test it out.

I like the idea of consolidating the document creation methods — I say let's delete `createDocumentWithId` entirely. Would you mind updating the README in this PR then, to reflect that that method isn't there anymore?

Again — looks great. I'll get back to you soon!

File: README.md [hunk 1]
```diff
@@ -49,7 +55,29 @@ firestore.updateDocument("FirstCollection/FirstDocument", data)
...
+If more specific queries need to be performed, you can use the `query` function followed by an `execute` invocation to get that data:
+
+```javascript
+const allDocumentsWithTest = firestore.query("FirstCollection").where("name", "==", "Test!").execute()
+```
+
+See other library methods and details [in the wiki](https://github.com/grahamearley/FirestoreGoogleAppsScript/wiki/Firestore-Method-Documentation).
+
+### Breaking Changes
+* v16: `createDocumentWithId(documentId, path, fields)` will be **removed** in the next update.
```
Comment (grahamearley) [inline README.md]: By this do you mean that method will be removed in v17?

Comment (LaughDonor) [reply README.md]: Why not? Unless you would like to just remove it immediately. I'm good either way.

Comment (grahamearley) [reply README.md]: Yeah, I'm leaning toward just removing it in this PR. I don't think this library is used at a large enough scale that this breaking change would wreak any havoc, so I say let's just do it. 

Good idea to call it out in the readme!

**Comment by LaughDonor** [comment]
All gone!

File: README.md [hunk 2]
```diff
@@ -37,9 +37,15 @@ const data = {
...
 
-Now, we can create a document called `FirstDocument` at a collection called `FirstCollection`:
+We can choose to create a document in collection called `FirstCollection` without an ID:
+
 ```javascript
-firestore.createDocumentWithId("FirstDocument", "FirstCollection", data)
+firestore.createDocument("FirstCollection", data)
+```
+
+Alternatively, we can create the document in the `FirstCollection` collection called called `FirstDocument`:
```
Comment (grahamearley) [inline README.md]: You have two "called"s in a row here :)

Comment (LaughDonor) [reply README.md]: :hushed:

**Comment by LaughDonor** [comment]
All cleaned up. Thanks for the checks!

File: Firestore.js [hunk 3]
```diff
@@ -26,92 +27,99 @@ var Firestore = function (email, key, projectId) {
...
    *
-   * @param {string} pathToCollection the path to the collection
-   * @return {object} an array of IDs of the documents in the collection
+   * @param {string} path the path of the document to update
+   * @param {object} fields the document's new fields
+   * @return {object} the Document object written to Firestore
    */
-  this.getDocumentIds = function (pathToCollection) {
-    return getDocumentIds_(pathToCollection, authToken, projectId)
+  this.updateDocument = function (path, fields) {
```
Comment (grahamearley) [inline Firestore.js]: `updateDocument` appears twice, and seems to be identical — copy paste error?

**Comment by grahamearley** [comment]
@LaughDonor Do you have any ideas on how to update our method documentation in this repo's wiki more automatically? I'd rather not have to keep manually typing it up when there are changes, but I'm not well-versed in Javascript, so I'm unaware of any nice auto-markdown or documentation generation tools. 

Any thoughts?

We'll have to get the wiki updated when this gets merged in and I release a new version.

File: Firestore.js (hunk 3 above)
Comment (LaughDonor) [reply Firestore.js]: Thanks, I was in preparation of the next PR, and forgot to undo. All fixed up @grahamearley

**Pull request merged**